xdsl/backend/jax_executable.py

```diff
@@ -0,0 +1,87 @@
+from collections.abc import Sequence
+from typing import Any
+
+import jax.numpy as jnp
+import numpy as np
+from jax import Array
+from jax._src import xla_bridge
+from jax._src.interpreters import mlir
+from jax._src.typing import SupportsDType
+from jaxlib.mlir import ir
+from jaxlib.xla_client import LoadedExecutable
+
+from xdsl.dialects.builtin import FunctionType, ModuleOp
+from xdsl.dialects.func import FuncOp
+from xdsl.traits import SymbolTable
+
+# JAX DTypeLike is currently broken
+# The np.dtype annotation in jax does not specify the generic parameter
+DTypeLike = (
+    str  # like 'float32', 'int32'
+    | type[Any]  # like np.float32, np.int32, float, int
+    | np.dtype[Any]  # like np.dtype('float32'), np.dtype('int32')
+    | SupportsDType  # like jnp.float32, jnp.int32
+)
+
+
+def array(object: Any, dtype: DTypeLike | None = None, copy: bool = True) -> Array:
+    """
+    Creates a jnp array with the passed-in parameters.
+
+    JAX type annotations are currently broken, as they don't provide a generic parameter to `np.dtype`.
+    This helper works around this issue.
+    """
+    return jnp.array(object, None, copy)  # pyright: ignore[reportUnknownMemberType]
+
+
+class JaxExecutable:
+    """
+    A class wrapping a jax LoadedExecutable.
+
+    Usage:
+
+    ``` python
+    my_module_str =\"""
+    func.func @main(%arg0: tensor<5xf32>) -> tensor<5xf32> {
+        ...
+    }
+    \"""
+    my_module = parser.parse_module(my_module_str)
+    @JaxExecutable.compile(module)
```

Comment:
I don't see a unit test using the decorator syntax nor a `__call__` that would delegate to `execute`. Is this intentional?

Reply:
Just saw the stacked PR which implements this. The documentation seems to just be ahead of its time :slightly_smiling_face:

Reply:
🔮

Reply:
I'll wait for both to be approved to merge this one